test/corpus/todo.txt

```diff
@@ -0,0 +1,486 @@
+      (identifier)
+      (constant))
+    (type_declaration
+      (identifier)
+      (constant)))
+  (enum_def
+    (constant)
+    (constant)
+    (constant)
+    (constant)
+    (constant))
+  (assign
+    (identifier)
+    (ERROR)
+    (constant))
+  (ERROR))
+
+==============
+pseudo methods
+==============
+
+class ClassName
+  def initialize
+    self.is_a?(Object)
+    self.nil?
+    typeof(self, other) # works
+    sizeof(self.class) # works
+    instance_sizeof(self.class) # works
+    pointerof(self)
+    offsetof(self.class, 1) # works
+    self.as(Object)
+    self.as?(Object)
+    responds_to?(:method)
+    self.!
+  end
+end
+
+---
+
+(source_file
+  (class_def
+    (constant)
+    (method_def
+      (identifier)
+      (call
+        (self)
+        (identifier)
+        (argument_list
+          (constant)))
+      (call
+        (self)
+        (identifier))
+      (typeof
+        (self)
+        (identifier))
+      (comment)
+      (sizeof
+        (class_type
+          (self)))
+      (comment)
+      (instance_sizeof
+        (class_type
+          (self)))
+      (comment)
+      (call
+        (identifier)
+        (argument_list
+          (self)))
+      (offsetof
+        (class_type
+          (self))
+        (integer))
+      (comment)
+      (call
+        (self)
+        (identifier)
+        (argument_list
+          (constant)))
+      (call
+        (self)
+        (identifier)
+        (argument_list
+          (constant)))
+      (call
+        (identifier)
+        (argument_list
+          (symbol)))
+      (call
+        (self)
+        (operator)))))
+
+=======================
+misc keywords / methods
+=======================
+
+super
+previous_def
+uninitialized Class
+
+---
+
+(source_file
+  (identifier)
+  (identifier)
+  (call
+    (identifier)
+    (argument_list
+      (constant))))
+
+====
+read instance var
+====
+
+object.@ivar
+
+---
+
+(source_file
+  (call
+    (identifier)
+    (MISSING identifier)
+    (argument_list
+      (instance_var))))
+
+
+===
+asm
+===
+
+asm(
+  # the assembly template string, following the
+  # syntax for LLVM's integrated assembler
+  "nop" :
+  # output operands
+  "=r"(foo), "=r"(bar) :
+  # input operands
+  "r"(1), "r"(baz) :
+  # names of clobbered registers
+  "eax", "memory" :
+  # optional flags, corresponding to the LLVM IR
+  # sideeffect / alignstack / inteldialect / unwind attributes
+  "volatile", "alignstack", "intel", "unwind"
+)
+
+---
+
+(source_file
+  (identifier)
+  (ERROR
+    (comment)
+    (comment)
+    (string))
+  (comment)
+  (chained_string
+    (string)
+    (ERROR
+      (identifier))
+    (string))
+  (ERROR
+    (identifier))
+  (comment)
+  (chained_string
+    (string)
+    (ERROR)
+    (string))
+  (ERROR
+    (identifier))
+  (comment)
+  (ERROR
+    (string))
+  (string)
+  (ERROR)
+  (comment)
+  (comment)
+  (ERROR
+    (string)
+    (string)
+    (string))
+  (string)
+  (ERROR))
+
+================
+methods as procs
```

Comment:
Also not a requirement but should this include block forwarding? (`&->`).

Reply:
These seem to work as-is

<img width="946" alt="image" src="https://github.com/user-attachments/assets/60329dff-3a1b-4e4e-aaab-705548ae112a">

Reply:
Oh nice 👍